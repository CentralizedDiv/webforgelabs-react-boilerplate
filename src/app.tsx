import React, { useMemo, useState, useCallback } from 'react';
import MainRouter from 'routes/main.router';
import { ConfigProvider, Layout } from 'antd';
import { hot } from 'react-hot-loader/root';
import STMenu from 'core/components/menu/menu.component';
import STTrigger from 'core/components/trigger/trigger.component';
import STHeader from 'core/components/header/header.container';
import { connect } from 'react-redux';
import { selectLayoutState } from 'core/core.reducer';
import { AppState } from 'redux/app-state';

import es_ES from 'antd/lib/locale/es_ES';
import en_US from 'antd/lib/locale/en_US';
import pt_BR from 'antd/lib/locale/pt_BR';

const { Header, Content, Sider } = Layout;
const browserLocale = window.navigator.language.toLocaleLowerCase();
const locale = browserLocale.search('pt') > -1 ? pt_BR : browserLocale.search('es') > -1 ? es_ES : en_US;

const STSider: React.FC = () => {
  const [collapsed, setCollapsed] = useState(true);
  const handleClick = useCallback(() => setCollapsed(!collapsed), [collapsed]);

  return useMemo(
    () => (
      <Sider
        className="st-sider"
        trigger={null}
        collapsible={true}
        collapsed={collapsed}
        collapsedWidth={64}
        width={250}
      >
        <STMenu />
        <STTrigger onClick={handleClick} collapsed={collapsed} />
      </Sider>
    ),
    [handleClick, collapsed]
  );
};

export interface IAppProps {
  readonly history: any;
  readonly showSider: any;
  readonly showHeader: any;
}

export const App: React.FC<IAppProps> = props => {
  return (
    <ConfigProvider locale={locale}>
      <Layout className="st-layout">
        {props.showSider && <STSider />}
        <Layout>
          {props.showHeader && (
            <Header className="st-header">
              <STHeader history={props.history} />
            </Header>
          )}
          <Content className="st-content">
            <MainRouter />
          </Content>
        </Layout>
      </Layout>
    </ConfigProvider>
  );
};

const mapStateToProps = (state: AppState) => selectLayoutState(state.uiState.global);

export default connect(mapStateToProps)(hot(App));
