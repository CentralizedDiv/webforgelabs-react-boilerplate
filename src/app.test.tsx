import React from 'react';
import { App, IAppProps } from './app';
import { shallow } from 'enzyme';

describe('App', () => {
  const props: IAppProps = {
    history: undefined,
    showHeader: false,
    showSider: false,
  };
  const component = shallow(<App {...props} />);

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

});