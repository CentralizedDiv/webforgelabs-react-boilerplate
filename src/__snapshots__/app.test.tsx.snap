// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<ConfigProvider
  locale={
    Object {
      "Calendar": Object {
        "lang": Object {
          "backToToday": "Back to today",
          "clear": "Clear",
          "dateFormat": "M/D/YYYY",
          "dateSelect": "select date",
          "dateTimeFormat": "M/D/YYYY HH:mm:ss",
          "dayFormat": "D",
          "decadeSelect": "Choose a decade",
          "month": "Month",
          "monthBeforeYear": true,
          "monthSelect": "Choose a month",
          "nextCentury": "Next century",
          "nextDecade": "Next decade",
          "nextMonth": "Next month (PageDown)",
          "nextYear": "Next year (Control + right)",
          "now": "Now",
          "ok": "Ok",
          "placeholder": "Select date",
          "previousCentury": "Last century",
          "previousDecade": "Last decade",
          "previousMonth": "Previous month (PageUp)",
          "previousYear": "Last year (Control + left)",
          "rangePlaceholder": Array [
            "Start date",
            "End date",
          ],
          "timeSelect": "select time",
          "today": "Today",
          "weekSelect": "Choose a week",
          "year": "Year",
          "yearFormat": "YYYY",
          "yearSelect": "Choose a year",
        },
        "timePickerLocale": Object {
          "placeholder": "Select time",
        },
      },
      "DatePicker": Object {
        "lang": Object {
          "backToToday": "Back to today",
          "clear": "Clear",
          "dateFormat": "M/D/YYYY",
          "dateSelect": "select date",
          "dateTimeFormat": "M/D/YYYY HH:mm:ss",
          "dayFormat": "D",
          "decadeSelect": "Choose a decade",
          "month": "Month",
          "monthBeforeYear": true,
          "monthSelect": "Choose a month",
          "nextCentury": "Next century",
          "nextDecade": "Next decade",
          "nextMonth": "Next month (PageDown)",
          "nextYear": "Next year (Control + right)",
          "now": "Now",
          "ok": "Ok",
          "placeholder": "Select date",
          "previousCentury": "Last century",
          "previousDecade": "Last decade",
          "previousMonth": "Previous month (PageUp)",
          "previousYear": "Last year (Control + left)",
          "rangePlaceholder": Array [
            "Start date",
            "End date",
          ],
          "timeSelect": "select time",
          "today": "Today",
          "weekSelect": "Choose a week",
          "year": "Year",
          "yearFormat": "YYYY",
          "yearSelect": "Choose a year",
        },
        "timePickerLocale": Object {
          "placeholder": "Select time",
        },
      },
      "Empty": Object {
        "description": "No Data",
      },
      "Icon": Object {
        "icon": "icon",
      },
      "Modal": Object {
        "cancelText": "Cancel",
        "justOkText": "OK",
        "okText": "OK",
      },
      "PageHeader": Object {
        "back": "Back",
      },
      "Pagination": Object {
        "items_per_page": "/ page",
        "jump_to": "Goto",
        "jump_to_confirm": "confirm",
        "next_3": "Next 3 Pages",
        "next_5": "Next 5 Pages",
        "next_page": "Next Page",
        "page": "",
        "prev_3": "Previous 3 Pages",
        "prev_5": "Previous 5 Pages",
        "prev_page": "Previous Page",
      },
      "Popconfirm": Object {
        "cancelText": "Cancel",
        "okText": "OK",
      },
      "Table": Object {
        "collapse": "Collapse row",
        "expand": "Expand row",
        "filterConfirm": "OK",
        "filterReset": "Reset",
        "filterTitle": "Filter menu",
        "selectAll": "Select current page",
        "selectInvert": "Invert current page",
        "sortTitle": "Sort",
      },
      "Text": Object {
        "copied": "Copied",
        "copy": "Copy",
        "edit": "Edit",
        "expand": "Expand",
      },
      "TimePicker": Object {
        "placeholder": "Select time",
      },
      "Transfer": Object {
        "itemUnit": "item",
        "itemsUnit": "items",
        "searchPlaceholder": "Search here",
        "titles": Array [
          "",
          "",
        ],
      },
      "Upload": Object {
        "previewFile": "Preview file",
        "removeFile": "Remove file",
        "uploadError": "Upload error",
        "uploading": "Uploading...",
      },
      "global": Object {
        "placeholder": "Please select",
      },
      "locale": "en",
    }
  }
>
  <Adapter
    className="st-layout"
  >
    <Adapter>
      <Adapter
        className="st-content"
      >
        <MainRouter />
      </Adapter>
    </Adapter>
  </Adapter>
</ConfigProvider>
`;
